### ISSUE TITLE
/PrintScaling

### ISSUE THREAD

**Issue by ousia** [sumatrapdfreader/sumatrapdf#98]
`/PrintScaling` when set to `None` (as described on [page 364 from the PDF spec](https://wwwimages2.adobe.com/content/dam/Adobe/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=364)) disables print scaling when printing the document.

This feature prints the document in real size and ignores the “fit to printer margins” option that slightly decreases the printed document size.

**Comment by kjk** [comment]
Unless there's evidence that this is useful and frequently used feature, we don't have plans to work on that.

**Comment by ousia** [comment]
@kjk, if this feature won’t be implemented, would it be possible to add a shortcut to launch _Acrobat_?

(I only need _Acrobat_ to read `/PrintScaling /None` and print accordingly, but having a faster way to load it would be helpful.)

Many thanks for your help.

**Comment by GitHubRulesOK** [comment]
@ousia 
SumatraPDF was supposed to add "Open in Acrobat" automatically to ExternalViewers list but it relies on an old registry setting that may not be present on your machine.
Thus it can be easy for me to set one of mine as a portable call
```
	[
		CommandLine = "F:\Apps\Adobe Readers\9\9.5.0_AdbeRdr950_en_US\Reader 9.0\Reader\AcroRd32.exe" /A page=%p "%1"
		Name = &Acrobat Reader for 3D Animations
		Filter = *.ai;*.pdf
	]
```

As you are probably aware the Adobe call to print starts something like -t
However SumatraPDF can only pass current filename or one page of current file (not a range) as arguments

**Comment by ousia** [comment]
@GitHubRulesOK,

many thanks for your fast reply.

I already have this, but I would like to have a keyboard shortcut, Is there a way to add a keyboard shortcut?

I would like to avoid using _Acrobat_, but I need to have this printer settings in _SumatraPDF_:

```
PrinterDefaults [
	PrintScale = shrink
]
```

I understand that the vast majority of users don’t know that there is such an option as `/PrintScale /None` (only #534 mentions it [I have just discovered]).

But it is really useful that the viewer can read and handle the `/ViewerPreferences` dictionary directly from the PDF document.

BTW, I think this may be more common than the use of annotation flags (such as in #1702).

**Comment by GitHubRulesOK** [comment]
the &A in my example is the same as if it were default in english <kbd>ALT</kbd><kbd>F</kbd><kbd>A</kbd> so for mine opens the current page in Acrobat Reader 9 Portable but I think it origionaly was A for Acrobat and R for Reader.

For command line I think -print-settings "noscale" = internal none

However "shrink" and "fit" should be same on command line as `shrink` or `fit` internally
The internal ones just preset the print dialog prior to users changing their preference and `shrink` should only generally be applied to oversize pages and `fit` to undersize ones but both can affect any size.

Print area "should be" as set in the WINDOWS DEFAULT  print driver so if set to border(margin)less then prints should be edge to edge.

**Comment by ousia** [comment]
@GitHubRulesOK, many thanks for your reply,

I really don’t get why this will not be implemented when there is a command line option for that (such as in #102).

**Comment by GitHubRulesOK** [comment]
@ousia 
Adobe have a large team that can quickly add features (not all good) to their PDF specification without regard to the onus that places on simpler viewers.
SumatraPDF as a minimalist PDF viewer application has never concentrated on those more optional pre-press directives that are most useful to print studios.

Printing has just too many options to allow for them all, and in effect SumatraPDF does not attempt to replace a good RIP which was the wayback intent of sending wrapped PS files to the printer engine that should handle those directives.

A good PDF printer should be able to accept a raw PS/PDF and process it internally using its own RIP. the concept was supposed to be as simple as Copy x.pdf > :lpt1 but then Adobe was born !

As a viewer SumatraPDF (via MuPDF backend) now only sends images of "as rendered" pages to Windows dib handler. The overhead of preparing each one of those pages in accordance with distiller embedded instructions is best handled by Adobe products that added them.

**Comment by ousia** [comment]
Many thanks for your reply, @GitHubRulesOK. I really appreciate it.

> @ousia
> Adobe have a large team that can quickly add features (not all good) to their PDF specification without regard to the onus that places on simpler viewers.

Well, I think this description is maybe missing a point: PDF is now a standard because _Adobe_ submitted it to the ISO, after 15 years of developing it as a format with its specification publicly available. (In fact, now is way harder to distribute the spec from the ISO than it was before [even _Adobe_ has noticed that].)

I’m afraid that even with free software such as _LibreOffice_ (or what it used to be, since its future status doesn’t seem extremely clear to me), I don’t think it would smart to consider the burden for external developments as a factor for feature adoption.

Sorry, but it wouldn’t make any sense (at least to me).

> SumatraPDF as a minimalist PDF viewer application has never concentrated on those more optional pre-press directives that are most useful to print studios.

There are other specification for printing studios. Duplex, page range, paper tray and scaling are simple features (all available for any printer, excepting duplex printing).

> Printing has just too many options to allow for them all, and in effect SumatraPDF does not attempt to replace a good RIP which was the wayback intent of sending wrapped PS files to the printer engine that should handle those directives.
> 
> A good PDF printer should be able to accept a raw PS/PDF and process it internally using its own RIP. the concept was supposed to be as simple as Copy x.pdf > :lpt1 but then Adobe was born !

I guess you mean _Acrobat_ (_Adobe_ created both PS and PDF, so it couldn’t be born after them).

> As a viewer SumatraPDF (via MuPDF backend) now only sends images of "as rendered" pages to Windows dib handler. The overhead of preparing each one of those pages in accordance with distiller embedded instructions is best handled by Adobe products that added them.

If I get your point, then having printing options in the command line for _SumatraPDF_ would make no sense.

This is what really strikes me. I could use the command line to select no page scaling, duplex mode and page to print in a graphical application for _Windows_.

I only use _Windows_ at work and I find _SumatraPDF_ way lighter than _Acrobat_. I work in a standard bussiness environment (no print studio or similar).

For me, having to use _Acrobat_ isn’t a problem at all. It might be slower, but I can avoid the issues that may derive from this. The real problem for me is recommending _SumatraPDF_. Let me explain this.

In my office, there is a group of 10 people that work generating PDF documents, browsing and printing them (among other things). About 5--6 hours a day, every 3--5 minutes. They complain about _Acrobat Reader DC_ being slow and unable to refresh when they regenerate a document (I haven’t checked the details, but it seems right to me).

I wanted to check whether _SumatraPDF_ would fit their needs. In their work, duplex printing comes from time to time and no print scaling is mandatory in many cases. Again, not in all of the documents. _Acrobat Reader_ is a lifesaver there, because they don’t need to think print options for each document. Just press Ctrl+P and Enter. After all, this is what computers are all about.

A different example, not directly related to _SumatraPDF_. I embed PDF documents in many of the PDF documents I generate at work. This is really useful for me and some of the people I work with thinks I’m using _Acrobat Pro_. I explained to one of them (who asked me first) that I use TeX for that. This is perfect for me, but it would be crazy for anyone else.

I’m also interested in providing a way to embed documents in PDF documents. _MuPDF_ comes to my mind. Their implementation of portfolio was problematic in many _Acrobat_ versions. They removed the support for this (instead of transforming it to `/EmbeddedFiles` as it was suggested to them).

Now version 1.17 contains `/FileAttachment` annotations. But only selecting the file to attach is a real pain, There are also privacy issues (the author is the user in _Windows_ [problematic in many companies]). And in the meantime, _MuPDF_ is unable to handle `/EmbeddedFiles`.

Do I need this for myself? Well, I don’t actually need them. I wouldn’t use _Acrobat Pro_ to attach files in PDF documents, because it would make my work way harder (than TeX does). And I can switch from _SumatraPDF_  back to _Acrobat Reader DC_ again.

But in both cases, I think this is a pity. Because I have to avoid suggesting some programs that could make their work easier.
